### ISSUE TITLE
ls: parsing "ls -lR"

### ISSUE THREAD

**Issue by vielmetti** [kellyjonbrazil/jc#23]
I don't get the expected results when I feed `ls -lR` into `jc`.

**Comment by kellyjonbrazil** [comment]
Hi there,

Yes, this is a known issue with the current release. ls is a bit difficult to parse due to its sometimes ambiguous output. I think we can support multiple directories and the recursive option with -l. I can start working on this for the next release.

Thanks for opening the issue!
kb

**Comment by vielmetti** [comment]
Thanks! Didn't realize that the output was ambiguous.

(Love this idea, by the way, hopefully as an interim step towards having these utilities generate this style of output directly)

**Comment by kellyjonbrazil** [comment]
I have globbing and -R working in the `dev` branch now. It's passing tests but if you'd like to test it  out that would be great.

You can clone the `dev` branch and then run the `install.sh` script to test:
```
git clone --single-branch --branch dev https://github.com/kellyjonbrazil/jc.git
```
then
```
cd jc
./install.sh
```

**Comment by vielmetti** [comment]
Installed and built without complaint (linux/arm64, for good measure).

The only thing that's a little funky is the "date" field, which I don't really know what to expect. The text matches up to what `ls` spits out (like "Feb 18 14:52"), which is correct but also not especially fun to parse. Do you have a recommended alternative for getting machine ready dates out of either `jc --ls` or some alternative? The other option is something derived from `find`.

**Comment by kellyjonbrazil** [comment]
Nice, thanks for testing!

Regarding the date, a couple of options. With `ls` you could use the `--time-style=full-iso` which seems to work fine with `jc`.

Another option would be to use the `stat` command which has all of the file time is full-iso format.

You can convert the date you get from 'ls' or 'stat' in Bash using the `date` command:
```
date -d"2020-02-18 02:55:37.967758487 +0000" +%s
```

**Comment by vielmetti** [comment]
Excellent. I also found that `ls --full-time` is equivalent to `ls --time-style=full-iso` (and now that I read the man page in detail it is clear there are is an infinite flexibility)

https://www.gnu.org/software/coreutils/manual/html_node/Formatting-file-timestamps.html

which can do any of the time stamp types as `date`

https://www.gnu.org/software/coreutils/manual/html_node/Time-conversion-specifiers.html#Time-conversion-specifiers

which means that I get a super easy-to-parse output with

```
ls -lR --time-style=+%s
```

but alas 

```
emv@q1:~$ jc -p ls -lR --time-style=+%s src/github.com/kellyjonbrazil/

jc:  Error - ls parser could not parse the input data. Did you use the correct parser?
     For details use the -d option.
```

with this long error

```
emv@q1:~$ jc -d -p ls -lR --time-style=+%s src/github.com/kellyjonbrazil/
Traceback (most recent call last):
  File "/home/emv/.local/bin/jc", line 11, in <module>
    load_entry_point('jc', 'console_scripts', 'jc')()
  File "/home/emv/src/github.com/kellyjonbrazil/jc/jc/cli.py", line 296, in main
    result = parser.parse(data, raw=raw, quiet=quiet)
  File "/home/emv/src/github.com/kellyjonbrazil/jc/jc/parsers/ls.py", line 255, in parse
    filename_field = parsed_line[8].split(' -> ')
IndexError: list index out of range
```

**Comment by kellyjonbrazil** [comment]
Yes, this is another instance of `ls` output being ambiguous, so the parser needs to guess the output. Unfortunately `ls` doesn't provide table headers so the parser figures out the field name by its position. I recommended the `full-iso` option because I was pleasantly surprised that it preserved the number of columns so it did not mess up the field parsing.

I would recommend using `full-iso` and then parsing the output with the `date` command. It would be very difficult to support any and all options for `ls` without some crazy regex tokenizing going on.

If the `-R` option is working for you now, I'll close this issue once I have merged it into the new release.

Thanks,
kb
